src/rdf4cpp/rdf/storage/util/StaticVec.hpp

```diff
@@ -0,0 +1,177 @@
+#ifndef RDF4CPP_STATICVEC_HPP
+#define RDF4CPP_STATICVEC_HPP
+
+#include <array>
+#include <cstddef>
+#include <stdexcept>
+
+namespace rdf4cpp::rdf::storage::util {
+
+/**
+ * A vector-like type with a capacity
+ * that is bounded by a compiletime constant.
+ * The data is stored in the StaticVec itself and not allocated on the heap.
+ *
+ * @tparam T element_type
+ * @tparam MAX_SIZE upper bound for the capacity (the vec cannot grow past this)
+ */
+template<typename T, size_t MAX_SIZE>
+struct StaticVec {
```

Comment:
This type was needed in an early version, but no longer needed. Do we want to keep it?

Reply:
I would remove it from this code base. The implementation looks nice, through. Please make a PR with it to this library: https://github.com/dice-group/dice-template-library

Reply:
:+1: